apps/arm/src/main.cpp

```diff
@@ -157,18 +157,42 @@ const static CANMsg::CANMsgHandlerMap canHandlerMap = {{HWBRIDGE::CANID::SET_TUR
 
 CANBus can1(CAN1_RX, CAN1_TX, HWBRIDGE::ROVERCONFIG::ROVER_CANBUS_FREQUENCY);
 
-// Incoming message processor
-void rxCANProcessor() {
-  CANMsg rxMsg;
+// CAN Processing Objects
+Mail<CANMsg, 100> mail_box;
+EventQueue event_queue;
 
+void rxCANClient() {
   while (true) {
-    if (can1.read(rxMsg)) {
-      canHandlerMap.at(rxMsg.getID())(rxMsg);  // TODO: handle failures
-    }
-    ThisThread::sleep_for(2ms);
+    CANMsg *mail = nullptr;
+    do {
+      mail = mail_box.try_get();  // TODO: try_get_for was not working. Investigate why and use it
+      ThisThread::sleep_for(1ms);
+    } while (mail == nullptr);
+    MBED_ASSERT((mail != nullptr));
+    canHandlerMap.at(mail->getID())(*mail);
+    MBED_ASSERT(mail_box.free(mail) == osOK);
```

Comment:
Thread should sleep here for 10 ms.

Reply:
Oh wait. Do the can can handle map functions have sleeps? I forgot we havent broken it into a seperate task yet

Reply:
Reason I say we should sleep here is cuz if we dont yield in the canhandlermap call, no lower prio threads have a chance to run (ex. Tx thread)

Reply:
Really need to enable mbed thread statistics to get to the bottom of some of these issues lol

Reply:
we yield inside the while loop. the functions in the map are quick. we can address this post sar